frontend/public/components/utils/tile-view-page.jsx

```diff
@@ -782,28 +863,49 @@ export class TileViewPage extends React.Component {
           <div className="co-catalog-page__header">
             <div className="co-catalog-page__heading text-capitalize">{activeCategory.label}</div>
             <div className="co-catalog-page__filter">
-              <FormControl
-                className="co-catalog-page__input"
-                type="text"
-                inputRef={(ref) => (this.filterByKeywordInput = ref)}
-                placeholder="Filter by keyword..."
-                bsClass="pf-c-form-control"
-                value={activeFilters.keyword.value}
-                onChange={(e) => this.onKeywordChange(e.target.value)}
-              />
+              <div>
+                <FormControl
+                  className="co-catalog-page__input"
+                  type="text"
+                  inputRef={(ref) => (this.filterByKeywordInput = ref)}
+                  placeholder="Filter by keyword..."
+                  bsClass="pf-c-form-control"
+                  value={activeFilters.keyword.value}
+                  onChange={(e) => this.onKeywordChange(e.target.value)}
+                />
+                {addGroupByDropdown && (
+                  <Dropdown
+                    className="co-catalog-page__btn-group__group-by"
+                    menuClassName="dropdown-menu--text-wrap"
+                    items={groupByTypes}
+                    onChange={(e) => this.onDropdownChange(e)}
+                    titlePrefix="Group By"
+                    title={activeFilters.groupBy.value || 'None'}
+                  />
+                )}
+              </div>
               <div className="co-catalog-page__num-items">{activeCategory.numItems} items</div>
             </div>
           </div>
-          {activeCategory.numItems > 0 && (
-            <Gallery
-              gutter="sm"
-              className="catalog-tile-view-pf catalog-tile-view-pf-no-categories co-catalog-page__grid"
-            >
-              {_.map(activeCategory.items, (item) => (
-                <GalleryItem key={`gallery-${item.uid}`}>{renderTile(item)}</GalleryItem>
-              ))}
-            </Gallery>
-          )}
+          {activeCategory.numItems > 0 &&
+            (activeFilters.groupBy.active ? (
+              this.renderGroupedItems(activeCategory.items, activeFilters.groupBy.value, renderTile)
+            ) : (
+              <Gallery
+                gutter="sm"
+                className="catalog-tile-view-pf catalog-tile-view-pf-no-categories co-catalog-page__grid"
```

Comment:
Move `co-catalog-page__grid` to a parent `div` so that this case and the grouping case use the same parent with this class.